From cc995d5396520659dca7cfb2439da2698118e6ae Mon Sep 17 00:00:00 2001
From: Emmanuel Jillela <emmanuel.jillela@intel.com>
Date: Mon, 2 May 2016 11:51:54 -0700
Subject: [PATCH 389/639] gpio: Fix debug show and clean up

Removed unnecessary debug prints, which may be annoying.
Fixed debug show  function which was printing wrong info for some pins.
Otherwise gpio sys kernel debug info will have wrong information.
---
 drivers/gpio/gpio-intelce2700.c | 13 ++++---------
 1 file changed, 4 insertions(+), 9 deletions(-)

diff --git a/drivers/gpio/gpio-intelce2700.c b/drivers/gpio/gpio-intelce2700.c
index f332853..732cfc0 100644
--- a/drivers/gpio/gpio-intelce2700.c
+++ b/drivers/gpio/gpio-intelce2700.c
@@ -754,11 +754,9 @@ static void intelce2700_writel(u32 value, void __iomem *reg)
 	unsigned long flags;
 
 	spin_lock_irqsave(&intelce2700_reg_access_lock, flags);
-	printk("GPIO write reg %x\n", value);
 	writel(value, reg);
 	/* simple readback to confirm the bus transferring done */
-//	readl(reg);
-	printk("GPIO write read back  %x\n", readl(reg));
+        //readl(reg);
 	spin_unlock_irqrestore(&intelce2700_reg_access_lock, flags);
 }
 
@@ -1126,10 +1124,7 @@ static void intelce2700_gpio_dbg_show(struct seq_file *s, struct gpio_chip *chip
 		label = gpiochip_is_requested(chip, i);
 		if (!label)
 			label = "Unrequested";
-
-		offs = FAMILY0_PAD_REGS_OFF +
-		      FAMILY_PAD_REGS_SIZE * (i / MAX_FAMILY_PAD_GPIO_NO) +
-		      GPIO_REGS_SIZE * (i % MAX_FAMILY_PAD_GPIO_NO);
+                offs = cg->pad_info[i].offset;
 
 		ctrl0 = intelce2700_readl(intelce2700_gpio_reg(chip, i, CE2700_GPIO_PADCTRL0_REG));
 		ctrl1 = intelce2700_readl(intelce2700_gpio_reg(chip, i, CE2700_GPIO_PADCTRL1_REG));
@@ -1201,7 +1196,7 @@ static void intelce2700_irq_shutdown(struct irq_data *d)
 }
 
 static struct irq_chip intelce2700_irqchip = {
-	.name		= "CHV-GPIO",
+	.name		= "CE2700-GPIO",
 	.irq_mask	= intelce2700_irq_mask,
 	.irq_unmask	= intelce2700_irq_unmask,
 	.irq_set_type	= intelce2700_irq_type,
@@ -1265,7 +1260,7 @@ static int intelce2700_gpio_irq_map(struct irq_domain *d, unsigned int virq,
 	struct intelce2700_gpio *cg = d->host_data;
 
 	irq_set_chip_and_handler_name(virq, &intelce2700_irqchip, handle_simple_irq,
-				      "demux");
+				      "ce2700-gpio-irq");
 	irq_set_chip_data(virq, cg);
 
 	return 0;
-- 
2.10.1

